src/app_updater.py

```diff
@@ -85,3 +99,146 @@ def _get_version_info_from_file(filepath: str) -> dict:
+        self.changes_made.append(change)
+
+    def _install_dependencies_from_requirements(self) -> None:
+        try:
+            subprocess.check_call(["pip", "install", "-r", self.requirements_path])
+            print("Dependencias instaladas con éxito.")
+        except subprocess.CalledProcessError:
+            print("Hubo un error al instalar las dependencias.")
+            traceback.print_exc()
+        except Exception as e:
+            print(f"Ocurrió un error inesperado: {e}")
+            traceback.print_exc()
+
+    def update_the_license_if_operator_wants(self):
+        user_messages = UserMessages(
+            prints=[],
+            input="Desea actualizar la app? [Y/n]",
+        )
+        if self._operator_wants(user_messages):
+            self._update_the_license()
+
+    def _update_the_license(self) -> None:
+
+        license = self._get_the_license_decoded()
+        license = self._update_expitation_date_if_operator_wants(license)
+        license = self._update_the_maximum_number_of_processes_if_operator_wants(license)
+        self._save_the_license_encoded(license)
+        self._log_change("Se actualizo la licencia:\n\n")
+        self._log_change(license)
+
+    def _get_the_license_decoded(self) -> str:
+        encodede_license = self._get_license_from_file()
+        decoded_license = self._decode_license(encodede_license)
+        return decoded_license
+
+    def _get_license_from_file(self) -> str:
+        with open(self.license_path, 'rb') as file:
+            encodede_license = file.read()
+        return encodede_license
+
+    def _decode_license(self, encodede_license: str) -> str:
+        decoded_license = base64.b64decode(encodede_license)
+        raw_yaml_license = decoded_license.decode('utf-8')
+        license = yaml.safe_load(raw_yaml_license)
+        return license
+
+    def _update_expitation_date_if_operator_wants(self, license: str) -> dict:
+        # TODO: refactor when the CLI class is created
+        expiration_date = self._get_expiration_date_from_license(license)
+        user_messages = UserMessages(
+            prints=[
+                f"La fecha de expiracion actual es: {expiration_date}",
+            ],
+            input="Desea actualizar la app? [Y/n]",
+        )
+        if self._operator_wants(user_messages):
+            self._update_expiration_date_asking_the_value_to_the_operator(license)
+
+        return license
+
+    def _get_expiration_date_from_license(self, license: str) -> str:
+        return license.get('license').get('expiration_date')
+
+    def _update_expiration_date_asking_the_value_to_the_operator(self, license: str) -> dict:
+        license['license']['expiration_date'] = self._get_new_expiration_date_from_operator()
+        return license
+
+    def _get_new_expiration_date_from_operator(self) -> str:
+        new_expiration_date = input("Ingrese la nueva fecha de expiracion: ")
+        return new_expiration_date
```

Comment:
Esto necesitará validar algun formato? 
Algo asi? esta medio cochinazo, pero habria que darle una vuelta, sobre todo porque estamos laburando con str y no datetime
```suggestion
        new_expiration_date = input("Ingrese la nueva fecha de expiracion (YYYY-MM-DD): ")
        try:
            new_expiration_date = datetime.datetime.strptime(date_string, '%Y-%m-%d')
        except ValueError:
            print("Formato incorrecto, debería ser YYYY-MM-DD")
            self._update_expitation_date_if_operator_wants(license)
         return new_expiration_date
```

Reply:
Me gusto la idea waalt, cree este ticket para trackearlo, asi voy mergeando esto mientras y queda planteada la mejora

https://www.notion.so/Hacer-validacion-de-la-fecha-ingresada-po-cli-por-el-operador-a9d2773a05aa4e6fa19126441964beaf?pvs=4